MSAL/test/unit/native_auth/public/MSALNativeAuthPublicClientApplicationTest.swift

```diff
@@ -364,4 +377,505 @@ final class MSALNativeAuthPublicClientApplicationTest: XCTestCase {
+        
+        let authResultFactoryMock = MSALNativeAuthResultFactoryMock()
+        let userAccountResult = MSALNativeAuthUserAccountResult(account: MSALNativeAuthUserAccountResultStub.account,
+                                                                authTokens: MSALNativeAuthTokens(accessToken: nil,
+                                                                                                 refreshToken: nil,
+                                                                                                 rawIdToken: nil),
+                                                                configuration: MSALNativeAuthConfigStubs.configuration,
+                                                                cacheAccessor: MSALNativeAuthCacheAccessorMock())
+        authResultFactoryMock.mockMakeUserAccountResult(userAccountResult)
+        
+        let tokenResult = MSIDTokenResult()
+        tokenResult.rawIdToken = "idToken"
+        let cacheAccessorMock = MSALNativeAuthCacheAccessorMock()
+        cacheAccessorMock.expectedMSIDTokenResult = tokenResult
+        
+        delegateSignInAfterSignUp.expectedUserAccountResult = userAccountResult
+        let signInAfterSignUpController = MSALNativeAuthSignInController(clientId: clientId,
+                                                                         signInRequestProvider: signInRequestProviderMock,
+                                                                         tokenRequestProvider: tokenRequestProviderMock, 
+                                                                         cacheAccessor: cacheAccessorMock,
+                                                                         factory: authResultFactoryMock,
+                                                                         signInResponseValidator: MSALNativeAuthSignInResponseValidatorMock(),
+                                                                         tokenResponseValidator: tokenResponseValidatorMock)
+        let signUpController = MSALNativeAuthSignUpController(config: configuration,
+                                                              requestProvider: signUpRequestProviderMock,
+                                                              responseValidator: signUpResponseValidatorMock,
+                                                              signInController: signInAfterSignUpController)
+        
+        let controllerFactory = MSALNativeAuthControllerRequestProviderFactoryMock(signUpController: signUpController)
+        
+        sut = MSALNativeAuthPublicClientApplication(
+            controllerFactory: controllerFactory,
+            inputValidator: MSALNativeAuthInputValidator(),
+            internalChallengeTypes: []
+        )
+        
+        // Correlation Id is validated internally against expectedStartRequestParameters and expectedChallengeRequestParameters
```

Comment:
Silviu, can you explain how this validates internally? I think I don't understand how this test works.

Reply:
By right clicking and jumping to the definition of the `expectedStartRequestParameters` and `expectedChallengeRequestParameters` in the `MSALNativeAuthSignUpRequestProviderMock` there are the functions `checkStartParameters` and `checkChallengeParameters` which make sure that the `context.correlationId` is equal to `expectedChallengeRequestParameters.context.correlationId`. 
Each of the `RequestProviderMock` has these checks as long as the expected Parameters are set. Some only check the context some check all parameters
![Screenshot 2023-11-24 at 10 01 44](https://github.com/AzureAD/microsoft-authentication-library-for-objc/assets/111577419/04052b20-1d9c-4249-88d9-245be6323448)
![Screenshot 2023-11-24 at 10 01 53](https://github.com/AzureAD/microsoft-authentication-library-for-objc/assets/111577419/604e0f9a-05b4-4c7b-8b8b-693afa0182f0)
I can specify maybe in the comments, "internally, inside the mocked Requests," if that clarifies it more. They are indeed rather hard to keep track and one of the reasons the unit tests took so long.

Reply:
Thank you, I understand now

Reply:
It would be nice to include where exactly the correlationId is validated instead of saying "internally", that would make debugging easier

Reply:
Done